registry/quilt_server/views.py

```diff
@@ -919,6 +919,17 @@ def _generate_preview(node, max_depth=PREVIEW_MAX_DEPTH):
     else:
         return None
 
+def _iterate_data_nodes(node):
+    # TODO: Merge into core.py
+    if isinstance(node, (TableNode, FileNode)):
+        yield node
+    elif isinstance(node, GroupNode):
+        for child in node.children.values():
+            yield from _iterate_data_nodes(child)
+
+# We don't know if file paths are Windows or UNIX... So just treat both / and \ as separators.
+FILE_EXT_RE = re.compile(r'\.([^/\\.]*)$')
```

Comment:
Then we'd have a problem if someone is crazy enough to run the registry on Windows :) (Yeah, it actually works.)

I realized that `pathlib.PureWindowsPath` handles both `\` and `/` (since both are legal separators on Windows... sort of). Would that be better?

Reply:
i played with `PurePath` and `PureWindowsPath` and seems to do what we want. i was kinda hoping that just `PurePath` was enough but i think you know those classes better than i do.